src/Fixer/FunctionNotation/NativeFunctionInvocationFixer.php

```diff
@@ -378,6 +378,7 @@ private function getAllCompilerOptimizedFunctionsNormalized(): array
             'is_string',
             'ord',
             'sizeof',
+            'sprintf',
```

Comment:
what about other functions like `vsprintf`?

Reply:
the optimization has only been implemented for `sprintf`. There is no optimization for `vsprintf` for now

Reply:
I have questioned this in https://github.com/php/php-src/pull/14546#issuecomment-2175948808

Reply:
My guess is that `vsprintf` is harder to transform into a concatenation because the compiler sees a single argument (the array of values) and has no easy way to destructure it at compile time.

Thus, to me, the use case for `vsprintf` compared to `sprintf` is likely to be the case where the array of values comes from a variable (using `vsprintf` with an array literal is just adding the overhead of creating an array and then destroying it, compared to `sprintf`). And the case of a variable would not be optimizable anyway.